pkg/resources/openstack/lb.go

```diff
@@ -17,27 +17,32 @@ limitations under the License.
 package openstack
 
 import (
-	"strings"
-
-	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/listeners"
 	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/loadbalancers"
-	v2pools "github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/pools"
+	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/monitors"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/subnets"
 	"k8s.io/klog"
 	"k8s.io/kops/pkg/resources"
 	"k8s.io/kops/upup/pkg/fi"
 	"k8s.io/kops/upup/pkg/fi/cloudup/openstack"
 )
 
 const (
-	typeLB  = "LoadBalancer"
-	typeLBL = "LBListener"
-	typeLBP = "LBPool"
+	typeLB   = "LoadBalancer"
+	typeLBL  = "LBListener"
+	typeLBP  = "LBPool"
+	typeLBPM = "LBPoolMonitor"
 )
 
-func (os *clusterDiscoveryOS) ListLB() ([]*resources.Resource, error) {
+func (os *clusterDiscoveryOS) DeleteSubnetLBs(subnet subnets.Subnet) ([]*resources.Resource, error) {
 	var resourceTrackers []*resources.Resource
+
+	if os.osCloud.UseOctavia() {
```

Comment:
@zetaab I am not overly familiar with octavia if you could give this a look.

Reply:
if using octavia we do not need delete any subresources (listener,monitor,pool,members). Instead we can just delete loadbalancer itself and use cascade option there. So most part of the code is useless if using octavia. Example when using octavia https://github.com/kubernetes/kops/blob/master/pkg/resources/openstack/lb.go#L53-L56

Reply:
Thanks ill update to reflect this.